### PR TITLE
Fix mutators not removing empty input

### LINKED ISSUE #1310: Undo bug in list constructor mutator
[From the forum](https://groups.google.com/d/msg/mitappinventortest/BI-wxJckMTk/TLJS7Tz2AgAJ): Create a list block with multiple mutators and values. In the mutator, remove all of the items. Undo the operation using Ctrl+Z (Cmd+Z). The block will have both "create empty list" and "make a list" inputs.

### PR DESCRIPTION
**Pull request by BeksOmega**
### Resolves
Closes #1310 

### Testing

I tested all of the mutators to confirm that they are now working/still working correctly!
procedure2
procedure1
intialize2
initialize1
dictionary
listadd
listcreate
join
bitwise
min
mult
add
if

### CODE REVIEW

**Comment by AppInventorWorkerBee** [comment]
Can one of the admins verify this patch?

**Comment by ewpatton** [comment]
@AppInventorWorkerBee ok to test

**Comment by ewpatton** [comment]
@BeksOmega Since this change only affects the blocks editor's behavior (nothing changes in the companion), you can have the base branch be `master` rather than `ucr`.

**Comment by BeksOmega** [comment]
Will fix!

But for my own future reference, what do changes to the companion include? My guess is: code generation (new blocks, and new block options) and new designer components?

(Sorry if you include this in docs somewhere, feel free to send me the link of shame)

File: appinventor/blocklyeditor/src/mutators.js [hunk 1]
```diff
@@ -41,8 +41,14 @@ Blockly.domToMutation = function(container) {
     }
   }
 
-  for (var x = 0; x < this.itemCount_; x++) {
-    this.removeInput(this.repeatingInputName + x);
+  if (this.itemCount_ == 0) {
+    if(this.emptyInputName != null) {
```
Comment (ewpatton) [inline appinventor/blocklyeditor/src/mutators.js]: Code style: space between `if` and `(`

**Comment by ewpatton** [comment]
> what do changes to the companion include?

The reason for the second branch is that companion updates are costly (time wise) both in terms of us testing and uploading to the Google Play Store, and then for everyone who needs to update to the new version. We tend to do them less frequently than other updates, so we have a separate branch (i.e., `ucr`) to collect changes to the companion. Changes for `ucr` include the following:

1. Any change that changes the behavior of the companion app, either a bug fix in an existing function or by adding or modifying a component's API (basically anything under `components/src/com/google/appinventor/components/runtime` and would include adding whole components). Exception: If a change only affects compiled apps, the change can go against `master` since it only requires updating the build servers, not the companion app.

2. Any change to the blocks language that requires a bump to `BLOCKS_LANGUAGE_VERSION`, even if the change doesn't directly change the companion (such as #2026).

    YaVersion.java is used to control how projects are upgraded from one version of App Inventor to the next. Projects in newer versions are not guaranteed to work in older versions. For example, your for each key value in dictionary block doesn't exist in the production system, but if someone made a project with that block (say on ai2-test) and then tried to load it elsewhere (say on ai2), then we can see that the version numbers don't align and can warn them in advance that it probably isn't going to work. In order to keep the ordering of version numbers monotonically increasing, they need to (ideally) be done on the same branch, which is why I moved the other branch to `ucr`.

In some rare instances, we will put a change against `master` where it makes sense and then merge `master` into `ucr`, even if the change might otherwise make sense to go against `ucr`. Example: #1998 adds a new block, so it goes against `ucr` (rule 2), but there was an issue with unit tests on Windows. I submitted #2044 against `master` since it affects both branches (but not directly the companion). After #2044 was merged, I merged `master` into `ucr` so that #1998 could be rebased on top to benefit from the fix. I could have just as easily submitted the change against `ucr`, but since we only ever merge `ucr` into `master` when we do a companion release, it was more beneficial to put the change on `master` and go the other way. 

This particular change doesn't meet the criteria for either rule 1 or 2, so it's safe to submit it against `master`. Technically, anything for `master` is also acceptable against `ucr`, but using `ucr` as the base branch means it won't go live unless we do a companion release, so generally it's better to opt for `master` if neither rule above applies.

**Comment by BeksOmega** [comment]
@ewpatton style has been fixed.

Also thank you for the explanation of which branch to work from! It makes a lot more sense now :D

**Pull request merged**